common/thorhelper/thorxmlwrite.cpp

```diff
@@ -685,6 +685,590 @@ StringBuffer &buildJsonFooter(StringBuffer  &footer, const char *suppliedFooter,
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    StringBuffer v;
+    v.append(field);
+    addContentField(v.str(), fieldName);
+}
+
+void CommonCSVWriter::outputDecimal(const void* field, unsigned size, unsigned precision, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    StringBuffer v;
+    char dec[50];
+    BcdCriticalBlock bcdBlock;
+    if (DecValid(true, size*2-1, field))
+    {
+        DecPushDecimal(field, size, precision);
+        DecPopCString(sizeof(dec), dec);
+        const char *finger = dec;
+        while(isspace(*finger)) finger++;
+        v.append(finger);
+    }
+    addContentField(v.str(), fieldName);
+}
+
+void CommonCSVWriter::outputUDecimal(const void* field, unsigned size, unsigned precision, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    StringBuffer v;
+    char dec[50];
+    BcdCriticalBlock bcdBlock;
+    if (DecValid(false, size*2, field))
+    {
+        DecPushUDecimal(field, size, precision);
+        DecPopCString(sizeof(dec), dec);
+        const char *finger = dec;
+        while(isspace(*finger)) finger++;
+        v.append(finger);
+    }
+    addContentField(v.str(), fieldName);
+}
+
+void CommonCSVWriter::outputUnicode(unsigned len, const UChar* field, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    StringBuffer v;
+    char * buff = 0;
+    unsigned bufflen = 0;
+    rtlUnicodeToCodepageX(bufflen, buff, len, field, "utf-8");
+    addStringField(bufflen, buff, fieldName);
+    rtlFree(buff);
+}
+
+void CommonCSVWriter::outputQString(unsigned len, const char* field, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    MemoryAttr tempBuffer;
+    char * temp;
+    if (len <= 100)
+        temp = (char *)alloca(len);
+    else
+        temp = (char *)tempBuffer.allocate(len);
+    rtlQStrToStr(len, temp, len, field);
+    addStringField(len, temp, fieldName);
+}
+
+void CommonCSVWriter::outputUtf8(unsigned len, const char* field, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    addStringField(rtlUtf8Size(len, field), field, fieldName);
+}
+
+void CommonCSVWriter::outputNumericString(const char* field, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    addStringField((size32_t)strlen(field), field, fieldName);
+}
+
+void CommonCSVWriter::outputBeginNested(const char* fieldName, bool simpleNested)
+{
+    if (!fieldName || !*fieldName)
+        return;
+
+    if (readingCSVHeader)
+    {
+        addCSVHeader(fieldName, NULL, true, simpleNested);
+        if (simpleNested) //ECL SET has only one column (parent name should be used as column name).
+            columnID++;
+        nestedHeaderLayerID++;
+        currentParentXPath.append(fieldName).append("/");
+        return;
+    }
+
+    if (!strieq(fieldName, "row"))
```

Comment:
Are you sure you can rely on the fieldname "row" not being used for other types of fields.  And is this processing post xpath, where a row might have another name?  i.e.

<Records>
   <Record>

Rather than
<Records>
  <Row>
?